sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryKeyVaultSecretReference.cs

```diff
@@ -10,7 +10,7 @@
 namespace Azure.Core.Expressions.DataFactory
 {
     /// <summary> Azure Key Vault secret reference. </summary>
-    [TypeReferenceType]
+    [TypeReferenceType(false, new[]{ nameof(SecretBaseType)})]
```

Comment:
The relevant code that would need to be tested is in the autorest repo (this [method](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/Decorator/PropertyMatchDetection.cs#L17)). Once this is merged we will be able to test it from the autorest side.

Reply:
Could we add more context to the PR description to explain what this change does?

Reply:
Sure, I tried to improve the docs for the attribute type so that it is more clear what the parameters do in general, but I will discuss the specific implications for this type as well.

Reply:
@christothes - I've updated the PR description.